### PR TITLE
Add page frame and settings navigation

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Add basic page layout. Fixes: https://github.com/canonical-web-and-design/MAAS-squad/issues/1330.
- Added basic settings nav.

QA:
- Load the UI. Notice the new header.
- Click on Settings.
- The header should change and you should get nav down the left (see https://app.zeplin.io/project/5ce5423a25863c1e0d7abb27/screen/5d1cbd18160fa1aff58546d0).
- Navigating to other settings pages (not General) should retain the heading/nav and highlight the page you're on.

### CODE REVIEW

File: ui/src/app/settings/components/Nav/Nav.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { HashLink } from "react-router-hash-link";
+import { Link } from "react-router-dom";
+import { withRouter } from "react-router";
+import classNames from "classnames";
+import React from "react";
+import ReactRouterPropTypes from "react-router-prop-types";
+
+import "./Nav.scss";
+
+function _generateSection(section, location) {
```
Comment (squidsoup) [inline ui/src/app/settings/components/Nav/Nav.js]: We don't tend to use the ES5 `function` keyword anymore. There's nothing wrong with it, but for consistency, can we use ES6 arrow functions exclusively?

File: ui/src/app/settings/containers/Repositories/Repositories.js [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
 
 import { RepositoryShape } from "../../proptypes";
 import actions from "../../actions";
 import selectors from "../../selectors";
 
-export class Repositories extends React.Component {
-  componentDidMount() {
-    this.props.fetchRepositories();
-  }
+export function Repositories(props) {
```
Comment (squidsoup) [inline ui/src/app/settings/containers/Repositories/Repositories.js]: as before, `export const Repositories = ({ fetchRepositories }) => {` is probably nicer.

**Comment by squidsoup** [comment]
Ah, thanks for fixing the routes in that commit :) +1

File: ui/src/app/base/components/Section/Section.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+
+import "./Section.scss";
+
+const Section = props => {
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/base/components/Section/Section.js]: Done.

File: ui/src/app/base/components/Section/Section.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <div className="section">
+      <header className="p-strip section__header">
+        <div className="row">
+          <h1 className="p-heading--four u-no-margin--bottom">{props.title}</h1>
+        </div>
+      </header>
+      <div className="p-strip u-no-padding">
+        <div className="row u-equal-height section__content-wrapper">
+          {props.sidebar ? (
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/base/components/Section/Section.js]: Done.

File: ui/src/app/settings/components/Nav/Nav.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/settings/components/Nav/Nav.js]: Done.

File: ui/src/app/settings/containers/Repositories/Repositories.js (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/settings/containers/Repositories/Repositories.js]: Done.

**Pull request merged**